build.gradle

```diff
@@ -1364,6 +1365,9 @@ project(':metadata') {
     testImplementation project(':raft').sourceSets.test.output
     testImplementation project(':server-common').sourceSets.test.output
 
+    testImplementation project(':test-common:test-common-api')
+    testRuntimeOnly project(':test-common:test-common-runtime')
```

Comment:
why only `metadata` module declares `test-common:test-common-runtime` as `testRuntimeOnly`?

Reply:
This was just done to illustrate the approach (and test it in the build). I'll revert this bit before merging